src/client/src/containers/AzureLoginModal/index.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import { connect } from "react-redux";
+
+import asModal from "../../components/Modal";
+
+import { injectIntl, InjectedIntlProps } from "react-intl";
+import { closeModalAction } from "../../actions/modalActions/modalActions";
+import { AppState } from "../../reducers";
+import { Dispatch } from "redux";
+import RootAction from "../../actions/ActionType";
+import { isAzureLoginModalOpenSelector } from "../../selectors/modalSelector";
+import { EXTENSION_COMMANDS, EXTENSION_MODULES } from "../../utils/constants";
+import buttonStyles from "../../css/buttonStyles.module.css";
+import styles from "./styles.module.css";
+import classnames from "classnames";
+import { FormattedMessage } from "react-intl";
+import keyUpHandler from "../../utils/keyUpHandler";
+import { strings as messages } from "./strings";
+import { KEY_EVENTS } from "../../utils/constants";
+import { ReactComponent as Cancel } from "../../assets/cancel.svg";
+import CollapsibleInfoBox from "../../components/CollapsibleInfoBox";
+
+interface IStateProps {
+  isModalOpen: boolean;
+  vscode: any;
+  isLoggedIn: boolean;
+}
+
+interface IDispatchProps {
+  closeModal: () => any;
+}
+
+type Props = IStateProps & IDispatchProps & InjectedIntlProps;
+
+const AzureLoginModal = (props: Props) => {
+  const { formatMessage } = props.intl;
+  const { isLoggedIn, closeModal } = props;
+  const handleSignInClick = () => {
+    // initiates a login command to VSCode ReactPanel class
+    props.vscode.postMessage({
+      command: EXTENSION_COMMANDS.AZURE_LOGIN,
+      module: EXTENSION_MODULES.AZURE,
+      track: true
+    });
+  };
+
+  const cancelKeyDownHandler = (event: React.KeyboardEvent<SVGSVGElement>) => {
+    if (event.key === KEY_EVENTS.ENTER || event.key === KEY_EVENTS.SPACE) {
+      event.preventDefault();
+      event.stopPropagation();
+      closeModal();
+    }
+  };
+
+  React.useEffect(() => {
+    // close sign in modal when user logs in
+    if (isLoggedIn) {
+      closeModal();
+    }
+  }, [isLoggedIn]);
+
+  return (
+    <div>
+      <div className={styles.headerContainer}>
+        <div className={styles.title}>
+          {formatMessage(messages.getStartedWithAzure)}
+        </div>
+        <Cancel
+          tabIndex={0}
+          className={styles.cancelIcon}
+          onClick={props.closeModal}
+          onKeyDown={cancelKeyDownHandler}
+        />
+      </div>
+
+      <div className={styles.questionaryContainer}>
+        <CollapsibleInfoBox
```

Comment:
Are these togglable via tab + enter? Just wondering.

Reply:
Not at the moment

Reply:
I added the tab in my PR :) now it is togglable via tab + enter